src/compute/src/logging/compute.rs

```diff
@@ -133,387 +135,150 @@ pub fn construct<A: Allocate>(
-                    // [here](https://github.com/MaterializeInc/materialize/pull/17302#discussion_r1086373740)
-                    // for more details, and think about this
-                    // again if we ever decide to stabilize it.
-                    u64::try_from(delay_sum).ok().into(),
-                ])
-            }
+        let frontier_current = frontier.as_collection().map(move |datum| {
+            Row::pack_slice(&[
+                Datum::String(&datum.export_id.to_string()),
+                Datum::UInt64(u64::cast_from(worker)),
+                Datum::MzTimestamp(datum.frontier),
+            ])
         });
-
-        let peek_current = peek.as_collection().map({
-            move |(peek, worker)| {
-                Row::pack_slice(&[
-                    Datum::Uuid(peek.uuid),
-                    Datum::UInt64(u64::cast_from(worker)),
-                    Datum::String(&peek.id.to_string()),
-                    Datum::MzTimestamp(peek.time),
-                ])
-            }
+        let import_frontier_current = import_frontier.as_collection().map(move |datum| {
+            Row::pack_slice(&[
+                Datum::String(&datum.export_id.to_string()),
+                Datum::String(&datum.import_id.to_string()),
+                Datum::UInt64(u64::cast_from(worker)),
+                Datum::MzTimestamp(datum.frontier),
+            ])
+        });
+        let frontier_delay = frontier_delay.as_collection().map(move |datum| {
+            Row::pack_slice(&[
+                Datum::String(&datum.export_id.to_string()),
+                Datum::String(&datum.import_id.to_string()),
+                Datum::UInt64(u64::cast_from(worker)),
+                Datum::UInt64(datum.delay_pow.try_into().expect("pow too big")),
+                Datum::Int64(datum.count.into()),
+                // [btv] This is nullable so that we can fill in `NULL` if it overflows. That would
+                // be a bit far-fetched, but not impossible to imagine. See discussion
+                // [here](https://github.com/MaterializeInc/materialize/pull/17302#discussion_r1086373740)
+                // for more details, and think about this again if we ever decide to stabilize it.
+                u64::try_from(datum.sum).ok().into(),
+            ])
+        });
+        let peek_current = peek.as_collection().map(move |datum| {
+            Row::pack_slice(&[
+                Datum::Uuid(datum.uuid),
+                Datum::UInt64(u64::cast_from(worker)),
+                Datum::String(&datum.id.to_string()),
+                Datum::MzTimestamp(datum.time),
+            ])
         });
-
-        // Duration statistics derive from the non-rounded event times.
         let peek_duration = peek_duration
             .as_collection()
-            .arrange_named::<RowSpine<_, _, _, _>>("Arranged timely peek_duration")
+            .arrange_core::<_, RowSpine<_, _, _, _>>(
+                Exchange::new(move |_| u64::cast_from(worker)),
```

Comment:
I argued that we should keep it. The issues I'd like to prevent is that the arrangement looks like it can be re-used like other arrangements, which is not true. To me, `Exchange::new(...)` adds more visual noise than `Pipeline`, making it more obvious what's happening. There is a slight overhead runtime overhead, but hopefully it's pretty minimal?

Reply:
No, we don't need it. @antiguru found it more readable that way. I think I'd find a `Pipeline` more readable as a no-op `Exchange`, but I'm not used to reading timely code, so I'll defer to him.

Reply:
I see, that makes sense, thanks! I also doubt that the runtime overhead would be of concern here.